plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -325,21 +331,35 @@ && predicateMatches(parquetPredicate, block, dataSource, descriptorsByPath, parq
         return new ReaderPageSource(parquetPageSource, readerProjections);
     }
 
-    public static Optional<org.apache.parquet.schema.Type> getParquetType(GroupType groupType, boolean useParquetColumnNames, HiveColumnHandle column)
+    private static ColumnIO getColumnIo(HiveColumnHandle column, MessageColumnIO messageColumn, MessageType fileSchema, ParquetColumnMapping columnMapping)
     {
-        if (useParquetColumnNames) {
-            return Optional.ofNullable(getParquetTypeByName(column.getBaseColumnName(), groupType));
-        }
-        if (column.getBaseHiveColumnIndex() < groupType.getFieldCount()) {
-            return Optional.of(groupType.getType(column.getBaseHiveColumnIndex()));
+        if (columnMapping == FIELD_ID) {
+            return lookupColumnById(messageColumn, column.getBaseHiveColumnIndex());
         }
+        String columnName = columnMapping == NAME ? column.getBaseColumnName() : fileSchema.getFields().get(column.getBaseHiveColumnIndex()).getName();
+        return lookupColumnByName(messageColumn, columnName);
+    }
+
+    private static Optional<org.apache.parquet.schema.Type> getParquetType(GroupType groupType, ParquetColumnMapping columnMapping, HiveColumnHandle column)
+    {
+        switch (columnMapping) {
+            case FIELD_ID:
+                return Optional.ofNullable(getParquetTypeById(column.getBaseHiveColumnIndex(), groupType));
+            case NAME:
+                return Optional.ofNullable(getParquetTypeByName(column.getBaseColumnName(), groupType));
+            case COLUMN_INDEX:
+                if (column.getBaseHiveColumnIndex() < groupType.getFieldCount()) {
+                    return Optional.of(groupType.getType(column.getBaseHiveColumnIndex()));
+                }
 
-        return Optional.empty();
+                return Optional.empty();
+        }
+        throw new IllegalArgumentException("Unsupported column mapping: " + columnMapping);
```

Comment:
nit: for consistency sake, use the `default` case in switch (as it is done in the rest of the codebase).

Reply:
+1

Reply:
We shouldn't use `default` in this case. See https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-default-clause-in-exhaustive-enum-based-switch-statements and #6883